src/jsonLogic.js

```diff
@@ -0,0 +1,108 @@
+import jsonLogic from 'json-logic-js';
+
+/**
+ * Parses the JSON schema to extract the advanced validation logic and returns a set of functionality to check the current status of said rules.
+ * @param {Object} schema - JSON schema node
+ * @param {Object} initialValues - form state
+ * @returns {Object}
+ */
+export function createValidationChecker(schema) {
+  const scopes = new Map();
+
+  function createScopes(jsonSchema, key = 'root') {
```

Comment:
I _think_ i prefer it to be nested inside? I think you'd get all the testing you want out of `createValidationChecker`, no?

Reply:
Just to bike-shed for a brief moment - You _would_ get the testing from `createValidationChecker`, but if that's the case, does it not make `createScopes` an unnecessary abstraction since it's only used once?

Reply:
i'm open to change but im being a dumbdumb here about the differences, what would you rewrite specifically? Could you share an example